code/game/objects/items/puzzle_pieces.dm

```diff
@@ -0,0 +1,132 @@
+/obj/item/keycard
+	name = "security keycard"
+	desc = "This feels like it belongs to a door."
+	icon = 'icons/obj/puzzle_small.dmi'
+	icon_state = "keycard"
+	force = 0
+	throwforce = 0
+	w_class = WEIGHT_CLASS_TINY
+	throw_speed = 1
+	throw_range = 7
+	var/puzzle_id = null
+
+//Two test keys for use alongside the two test doors.
+/obj/item/keycard/cheese
+	name = "cheese keycard"
+	desc = "Look, I still don't understand the reference. What the heck is a keyzza?"
+	color = "#f0da12"
+	puzzle_id = "cheese"
+
+/obj/item/keycard/swordfish
+	name = "titanic keycard"
+	desc = "Smells like it was at the bottom of a harbor."
+	color = "#3bbbdb"
+	puzzle_id = "swordfish"
+
+//***************
+//*****Doors*****
+//***************
+
+/obj/machinery/door/keycard
+	name = "locked door"
+	desc = "This door only opens when a keycard is swiped. This thing looks virtually indestructable, however."
+	icon = 'icons/obj/doors/doorpuzzle.dmi'
+	icon_state = "door_closed"
+	explosion_block = 3
+	heat_proof = TRUE
+	max_integrity = 600
+	armor = list("melee" = 100, "bullet" = 100, "laser" = 100, "energy" = 100, "bomb" = 100, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 100)
+	resistance_flags = INDESTRUCTIBLE | FIRE_PROOF | ACID_PROOF | LAVA_PROOF
+	damage_deflection = 70
+	var/puzzle_id = null	//Make sure that the key has the same puzzle_id as the keycard door!
+
+//Standard Expressions to make keycard doors basically un-cheeseable
+/obj/machinery/door/keycard/Bumped(atom/movable/AM)
+	return !density && ..()
+
+/obj/machinery/door/keycard/emp_act(severity)
+	return
+
+/obj/machinery/door/keycard/ex_act(severity, target)
+	return
+
+/obj/machinery/door/keycard/try_to_activate_door(mob/user)
+	add_fingerprint(user)
+	if(operating)
+		return
+
+/obj/machinery/door/keycard/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(istype(I,/obj/item/keycard))
+		var/obj/item/keycard/key = I
+		if((!puzzle_id || puzzle_id == key.puzzle_id)  && density)
+			to_chat(user, "<span class='notice'>The door beeps, and slides opens.</span>")
+			open()
+			return
+		else if(puzzle_id != key.puzzle_id)
+			to_chat(user, "<span class='notice'>[src] buzzes. This must not be the right key.</span>")
+			return
+		else
+			to_chat(user, "<span class='notice'>This door doesn't appear to close.</span>")
+			return
+
+//Test doors. Gives admins a few doors to use quickly should they so choose.
+/obj/machinery/door/keycard/cheese
+	name = "blue airlock"
+	desc = "Smells like... pizza?"
+	puzzle_id = "cheese"
+
+/obj/machinery/door/keycard/swordfish
+	name = "blue airlock"
+	desc = "If nautical notions be something you wish."
+	puzzle_id = "swordfish"
+
+//*************************
+//***Box Pushing Puzzles***
+//*************************
+//We're working off a subtype of pressureplates, which should work just a BIT better now.
+/obj/structure/holobox
+	name = "holobox"
+	desc = "A projected hologram similar to PAi holochassis, containing a secure decryption key."
```

Comment:
```suggestion
	desc = "A hard-light crate containing a secure decryption key."
```

Reply:
I will defend that I don't want to describe/call it a crate, because I don't want to give anyone the idea that it has crate functionality (Burning it open to access the contents, Dragging to hop over, etc.)